spec/models/inning_spec.rb

```diff
@@ -0,0 +1,138 @@
+require "rails_helper"
+
+RSpec.describe Inning do
+  let(:game) { create(:scheduled_game) }
+
+  subject do
+    Inning.new(
+      game: game,
+      half: :top,
+      status: :completed,
+      number: 1,
+      runs: 0
+    )
+  end
+
+  it { should belong_to(:game) }
+
+  it { should define_enum_for(:half).with(%i(top bottom)) }
+  it { should define_enum_for(:status).with(%i(in_progress completed)) }
+
+  it { should validate_presence_of(:game) }
+  it { should validate_presence_of(:number) }
+  it { should validate_presence_of(:half) }
+  it { should validate_presence_of(:status) }
+
+  it { should validate_numericality_of(:number).is_greater_than_or_equal_to(1) }
+  it { should validate_numericality_of(:runs).is_greater_than_or_equal_to(0) }
+
+  describe "game/number/half uniqueness constraint" do
+    context "when the combination is unique" do
+      it "saves the record" do
+        expect { subject.save! }.to_not raise_error
+      end
+    end
+
+    context "when the combination is not unique" do
+      let(:duplicate_record) {
```

Comment:
This test only exists because I can't find a way for the shoulda matchers to handle this without upcasing the enum. It may be worth one more look before we go with a custom context.